### PR TITLE
Add an option for defining number of visible items in a NativeSelect

### PR DESCRIPTION
**Pull request by Artur-**
This change is [Reviewable](https://reviewable.io/reviews/vaadin/framework/9109)

### CODE REVIEW

File: server/src/main/java/com/vaadin/ui/NativeSelect.java [hunk 1]
```diff
@@ -203,4 +203,25 @@ public void setEmptySelectionCaption(String caption) {
...
+
+    /**
+     * Sets the number of items that are visible. If only one item is visible,
+     * then the box will be displayed as a drop-down list (the default).
+     *
+     * @param visibleItemCount
+     *            the visible item count
+     */
+    public void setVisibleItemCount(int visibleItemCount) {
+        getState().visibleItemCount = visibleItemCount;
```
Comment (tsuoanttila) [inline server/src/main/java/com/vaadin/ui/NativeSelect.java]: Check for negative values, throw illegal argument and update javadocs accordingly.

Comment (Artur-) [inline server/src/main/java/com/vaadin/ui/NativeSelect.java]: Done

**Comment by tsuoanttila** [comment]
:lgtm:

---

Reviewed 4 of 5 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9109#-:-Ki5SBGGdP8pgGP0ngyE:bnfp4nl)*

**Comment by hesara** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[client/src/main/java/com/vaadin/client/ui/VNativeSelect.java, line 119 at r2](https://reviewable.io:443/reviews/vaadin/framework/9109#-Ki5UWUvUpHO_MD-CiSG:-Ki5UWUvUpHO_MD-CiSH:b-z7mpbq) ([raw file](https://github.com/vaadin/framework/blob/060aeb90eec5ebfeadd2711feba3fd1c78bf53ce/client/src/main/java/com/vaadin/client/ui/VNativeSelect.java#L119)):*
> ```Java
>      *
>      * @param visibleItemCount
>      *            the visible item count
> ```

@since on the public methods

---

*[server/src/main/java/com/vaadin/ui/NativeSelect.java, line 214 at r2](https://reviewable.io:443/reviews/vaadin/framework/9109#-Ki5TzpWdzmYWqX4XfRl:-Ki5TzpXwE0EMyJsy_x0:b-z7mpbq) ([raw file](https://github.com/vaadin/framework/blob/060aeb90eec5ebfeadd2711feba3fd1c78bf53ce/server/src/main/java/com/vaadin/ui/NativeSelect.java#L214)):*
> ```Java
>      *            the visible item count
>      * @throws IllegalArgumentException
>      *             if the value is smaller than one
> ```

@since on the public methods

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9109)*

**Comment by hesara** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9109)*

**Comment by Artur-** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[client/src/main/java/com/vaadin/client/ui/VNativeSelect.java, line 119 at r2](https://reviewable.io:443/reviews/vaadin/framework/9109#-Ki5UWUvUpHO_MD-CiSG:-KiFj78pmpjYQ7PvOR-j:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/060aeb90eec5ebfeadd2711feba3fd1c78bf53ce/client/src/main/java/com/vaadin/client/ui/VNativeSelect.java#L119)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

@since on the public methods
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9109)*

**Comment by hesara** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9109)*

**Pull request merged**